pkg/rollout/rollout.go

```diff
@@ -96,36 +102,60 @@ func (r *Rollout) UpdateService(svc *run.Service) (*run.Service, error) {
 		r.log.Info("could not determine stable revision")
 		return nil, nil
 	}
-	r.log.Debugf("%q is the stable revision", stable)
+	r.log = r.log.WithField("stable", stable)
 
 	candidate := DetectCandidateRevisionName(svc, stable)
 	if candidate == "" {
 		r.log.Info("could not determine candidate revision")
 		return nil, nil
 	}
-	r.log.Debugf("%q is the candidate revision", candidate)
+	r.log = r.log.WithField("candidate", candidate)
+
+	// A new candidate does not have metrics yet, so it can't be diagnosed.
+	if isNewCandidate(svc, candidate) {
+		r.log.Debug("candidate is new, will assign it some traffic")
+		svc = r.PrepareRollForward(svc, stable, candidate)
+		svc = r.updateAnnotations(svc, stable, candidate)
+		return r.replaceService(svc)
+	}
 
-	svc = r.SplitTraffic(svc, stable, candidate)
-	svc = r.updateAnnotations(svc, stable, candidate)
-	svc, err := r.runClient.ReplaceService(r.project, r.serviceName, svc)
+	diagnosis, err := r.diagnoseCandidate(candidate, r.strategy.Metrics)
 	if err != nil {
-		return nil, errors.Wrapf(err, "could not update service %q", r.serviceName)
+		r.log.Error("could not diagnose candidate's health")
+		return nil, errors.Wrapf(err, "failed to diagnose health for candidate %q", candidate)
 	}
-	r.log.Debug("service succesfully updated")
 
-	return svc, nil
+	switch diagnosis.OverallResult {
+	case health.Inconclusive:
+		r.log.Debug("candidate does not have enough requests to determine health, skipping rollout/rollback for now")
```

Comment:
too detailed. try this:

`health check inconclusive`

realistically this is a debug so it's for devs